src/main/kotlin/cacophony/parser/LLOneParser.kt

```diff
@@ -1,15 +1,65 @@
 package cacophony.parser
 
 import cacophony.grammars.AnalyzedGrammar
+import cacophony.grammars.DFAStateReference
 import cacophony.grammars.ParseTree
+import cacophony.grammars.Production
 import cacophony.utils.Diagnostics
 
-class LLOneParser<SymbolType : Enum<SymbolType>> : Parser<SymbolType> {
+class LLOneParser<StateType, SymbolType : Enum<SymbolType>>(
+    private val nextAction: Map<SymbolType, MutableMap<DFAStateReference<StateType, SymbolType, Production<SymbolType>>, SymbolType?>>,
+) : Parser<SymbolType> {
     companion object {
-        fun <StateType, SymbolType : Enum<SymbolType>> fromAnalyzedGrammar(
+        // Constructs LLOneParser with computed nextAction map.
+        // nextAction gives symbol on the production that we should use next
+        // depending on the current input symbol and dfa state.
+        // null denotes no suitable production, then we should accept.
+        // Throws if grammar is not LL(1).
+        inline fun <StateType, reified SymbolType : Enum<SymbolType>> fromAnalyzedGrammar(
             analyzedGrammar: AnalyzedGrammar<StateType, SymbolType>,
-        ): LLOneParser<SymbolType> {
-            TODO("Not yet implemented")
+        ): LLOneParser<StateType, SymbolType> {
+            val nextAction: Map<SymbolType, MutableMap<DFAStateReference<StateType, SymbolType, Production<SymbolType>>, SymbolType?>> =
+                enumValues<SymbolType>().toList().associateWith { mutableMapOf() }
+
+            analyzedGrammar.automata.forEach { (dfaLabel, dfa) -> // automaton for the grammar symbol
```

Comment:
`dfaLabel` is not used, is it fine? If so, you can change this line to `analyzedGrammar.automata.forEach { (_, dfa)`